source/tomopy/recon/rotation.py

```diff
@@ -250,49 +256,111 @@ def find_center_vo(tomo, ind=None, smin=-50, smax=50, srad=6, step=0.25,
+    height_dsp = height // dsp_fact0
+    width_dsp = width // dsp_fact1
+    image_dsp = image[0:dsp_fact0 * height_dsp, 0:dsp_fact1 * width_dsp]
+    image_dsp = image_dsp.reshape(
+        height_dsp, dsp_fact0, width_dsp, dsp_fact1).mean(-1).mean(1)
+    return image_dsp
+
+
+def _calculate_metric(shift_col, sino1, sino2, sino3, mask):
+    """
+    Metric calculation.
+    """
+    shift_col = 1.0 * np.squeeze(shift_col)
+    if np.abs(shift_col - np.floor(shift_col)) == 0.0:
+        shift_col = int(shift_col)
+        sino_shift = np.roll(sino2, shift_col, axis=1)
+        if shift_col >= 0:
+            sino_shift[:, :shift_col] = sino3[:, :shift_col]
+        else:
+            sino_shift[:, shift_col:] = sino3[:, shift_col:]
+        mat = np.vstack((sino1, sino_shift))
+    else:
+        sino_shift = ndimage.interpolation.shift(
+            sino2, (0, shift_col), order=3, prefilter=True)
+        if shift_col >= 0:
+            shift_int = int(np.ceil(shift_col))
+            sino_shift[:, :shift_int] = sino3[:, :shift_int]
+        else:
+            shift_int = int(np.floor(shift_col))
+            sino_shift[:, shift_int:] = sino3[:, shift_int:]
+        mat = np.vstack((sino1, sino_shift))
+    metric = np.mean(
+        np.abs(np.fft.fftshift(fft2(mat))) * mask)
+    return np.asarray([metric], dtype=np.float32)
+
+
+def _search_coarse(sino, smin, smax, ratio, drop, parallel=True):
     """
     Coarse search for finding the rotation center.
     """
     (nrow, ncol) = sino.shape
     cen_fliplr = (ncol - 1.0) / 2.0
     smin = np.int16(np.clip(smin + cen_fliplr, 0, ncol - 1) - cen_fliplr)
     smax = np.int16(np.clip(smax + cen_fliplr, 0, ncol - 1) - cen_fliplr)
-    # Flip left-right the [0:Pi ] sinogram to make a full [0;2Pi] sinogram
     flip_sino = np.fliplr(sino)
-    # Below image is used for compensating the shift of the [Pi;2Pi] sinogram
-    # It helps to avoid local minima.
     comp_sino = np.flipud(sino)
     list_shift = np.arange(smin, smax + 1)
-    list_metric = np.zeros(len(list_shift), dtype='float32')
     mask = _create_mask(2 * nrow, ncol, 0.5 * ratio * ncol, drop)
-    sino_sino = np.vstack((sino, flip_sino))
-    abs_fft2_sino = np.empty_like(sino_sino)
-    for i in list_shift:
-        _sino = sino_sino[nrow:]
-        _sino[...] = np.roll(flip_sino, i, axis=1)
-        if i >= 0:
-            _sino[:, 0:i] = comp_sino[:, 0:i]
-        else:
-            _sino[:, i:] = comp_sino[:, i:]
-        fft2sino = np.fft.fftshift(fft2(sino_sino))
-        np.abs(fft2sino, out=abs_fft2_sino)
-        abs_fft2_sino *= mask
-        list_metric[i - smin] = abs_fft2_sino.mean()
+    if parallel is True:
+        list_metric = distribute_jobs(np.float32(list_shift),
+                                      _calculate_metric, axis=0,
+                                      args=(sino, flip_sino, comp_sino, mask),
+                                      ncore=mp.cpu_count() - 1, nchunk=1)
+    else:
```

Comment:
For example, instead of setting `parallel=False`, users can set `ncore=1` and `nchunk=1`.